app/code/Magento/Catalog/Helper/Product/ProductList.php

```diff
@@ -45,15 +38,21 @@ class ProductList
     protected $_defaultAvailableLimit = [10 => 10, 20 => 20, 50 => 50];
 
     /**
+     * @var ToolbarConfig
+     */
+    private $toolbarConfig;
+
+    /**
+     * ProductList constructor.
      * @param ScopeConfigInterface $scopeConfig
-     * @param Registry $coreRegistry
+     * @param ToolbarConfig $toolbarConfig
      */
     public function __construct(
         ScopeConfigInterface $scopeConfig,
-        Registry $coreRegistry = null
+        ToolbarConfig $toolbarConfig
```

Comment:
This class marked as `@api`, so we can't just replace existing argument with a new one. Please introduce new dependecy as optional argument